src/shapes/Group.ts

```diff
@@ -385,17 +313,15 @@ export class Group extends createCollectionMixin(
    * @param {FabricObject} object
    */
   _watchObject(watch: boolean, object: FabricObject) {
-    const directive: 'on' | 'off' = watch ? 'on' : 'off';
     //  make sure we listen only once
     watch && this._watchObject(false, object);
-    // @ts-expect-error TS limitations
-    object[directive]('changed', this.__objectMonitor);
-    // @ts-expect-error TS limitations
-    object[directive]('modified', this.__objectMonitor);
-    // @ts-expect-error TS limitations
-    object[directive]('selected', this.__objectSelectionTracker);
-    // @ts-expect-error TS limitations
-    object[directive]('deselected', this.__objectSelectionDisposer);
+    if (watch) {
+      object.on('selected', this.__objectSelectionTracker);
+      object.on('deselected', this.__objectSelectionDisposer);
```

Comment:
we probably need internals on/off to avoid the developer can kill those events without understanding how, like doing an object.off('deselected').
Not scope of this pr.

Reply:
I agree, we can add an options object with a flag for that.
Trurth is I think this bit can be removed

Reply:
in general we use a lot of internal .on/off.
We should just have a sort of _on and _off that we can setup and can't be broken from outside doing a simple off() that kills everything

Reply:
True
Specifically fir group I think these can be removed